packages/appstore/src/hooks/use-tradershub-tracking.ts

```diff
@@ -0,0 +1,113 @@
+import { Analytics } from '@deriv/analytics';
+import { useStore } from '@deriv/stores';
+import { useCallback, useEffect, useMemo } from 'react';
+
+// This hook is used to track the onboarding form in TradersHub
+export const useTradersHubTracking = () => {
+    const { traders_hub, ui, client } = useStore();
+
+    const { is_mobile } = ui;
+
+    const { loginid } = client;
+
+    const { is_first_time_visit } = traders_hub;
+
+    const form_source = useMemo(
+        () => (is_first_time_visit ? 'tradershub_first_entrance' : 'tradershub_dashboard_form'),
+        [is_first_time_visit]
+    );
+
+    const event_name = 'ce_tradershub_onboarding_form';
+
+    useEffect(() => {
+        Analytics.setAttributes({
+            device_type: is_mobile ? 'mobile' : 'desktop',
+            account_type: loginid?.slice(0, 2),
+        });
+    }, [is_mobile, loginid]);
+
+    const trackDotNavigation = useCallback(
+        (step: number) => {
+            Analytics.trackEvent(event_name, {
+                action: 'choose_step_navigation',
+                form_source,
+                step_num: step,
+                step_codename: `${step}_step`,
+            });
+        },
+        [form_source]
+    );
+
+    const trackLastStep = useCallback(() => {
+        Analytics.trackEvent('ce_tradershub_onboarding_form', {
```

Comment:
can we reuse the `event_name` variable in all of the places?

Reply:
I totally missed that one, thanks for pointing out 👍